compiler-core/src/manifest.rs

```diff
@@ -247,6 +262,10 @@ impl Default for ManifestPackage {
 pub enum ManifestPackageSource {
     #[serde(rename = "hex")]
     Hex { outer_checksum: Base16Checksum },
+    #[serde(rename = "git")]
+    Git { repo: String, commit: String },
+    #[serde(rename = "local")]
+    Local { path: PathBuf }, // should be the canonical path
```

Comment:
What does this comment mean?

Reply:
By the time this path reaches the manifest it should be in canonical format. At one point this was important, but I don't think it is any more. 

Nightly has a type for canonical paths, but I might just make a newtype to fill the gap.

Reply:
I ended up adding a test for this instead. Its more important that paths are absolute.